Sources/AblyChat/DefaultTyping.swift

```diff
@@ -39,165 +46,142 @@ internal final class DefaultTyping: Typing {
+                            await subscription.emit(
+                                TypingEvent(currentlyTyping: typingTimerManager.currentlyTypingClients(), change: .init(clientId: messageClientID, type: .stopped))
+                            )
                         }
                     }
-                    logger.log(message: "Failed to fetch presence set after \(maxRetryDuration) seconds. Giving up.", level: .error)
+
+                    // (CHA-T13) When a typing event (typing.start or typing.stop) is received from the realtime client, the Chat client shall emit appropriate events to the user.
+                    await subscription.emit(
+                        TypingEvent(
+                            currentlyTyping: typingTimerManager.currentlyTypingClients(),
+                            change: .init(clientId: messageClientID, type: .started)
+                        )
+                    )
                 }
             }
 
-            subscription.addTerminationHandler { [weak self] in
-                if let eventListener {
-                    self?.featureChannel.channel.presence.unsubscribe(eventListener)
+            let stoppedEventListener = featureChannel.channel.subscribe(TypingEvents.stopped.rawValue) { [weak self] message in
+                guard let self, let messageClientID = message.clientId else {
+                    return
                 }
-            }
 
-            return subscription
-        }
+                logger.log(message: "Received stopped typing message: \(message)", level: .debug)
 
-        // (CHA-T2) Users may retrieve a list of the currently typing client IDs. The behaviour depends on the current room status, as presence operations in a Realtime Client cause implicit attaches.
-        internal func get() async throws(ARTErrorInfo) -> Set<String> {
-            do throws(InternalError) {
-                logger.log(message: "Getting presence", level: .debug)
-
-                // CHA-T2c to CHA-T2f
-                do {
-                    try await featureChannel.waitToBeAbleToPerformPresenceOperations(requestedByFeature: RoomFeature.presence)
-                } catch {
-                    logger.log(message: "Error waiting to be able to perform presence get operation: \(error)", level: .error)
-                    throw error
+                Task {
+                    // (CHA-T13b4) If the event represents a client that has stopped typing, then the chat client shall remove the clientId from the typing set and emit the updated set to any subscribers. It shall also cancel the (CHA-T13b1) timeout for the typing client.
+                    await typingTimerManager.cancelTypingTimer(for: messageClientID)
+
+                    // (CHA-T13) When a typing event (typing.start or typing.stop) is received from the realtime client, the Chat client shall emit appropriate events to the user.
+                    await subscription.emit(
```

Comment:
What about CHA-T13b5, which is meant to ignore the event if the user was not typing?